stream.go

```diff
@@ -239,75 +247,114 @@ func (s *Stream) sendReset() error {
 
 // Reset resets the stream (forcibly closes the stream)
 func (s *Stream) Reset() error {
+	sendReset := false
 	s.stateLock.Lock()
 	switch s.state {
-	case streamInit:
-		// No need to send anything.
-		s.state = streamReset
-		s.stateLock.Unlock()
-		return nil
-	case streamClosed, streamReset:
+	case streamFinished:
 		s.stateLock.Unlock()
 		return nil
+	case streamInit:
```

Comment:
is this intentional (ie we do nothing) or a missing fallthrough?
I would add a comment or a break here if it's the former.

Reply:
It's an intentional do nothing. I'll add a comment.